provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/contract/http/ContractAPIHttp.java

```diff
@@ -126,6 +127,7 @@ private HttpRequest prepareRequest(
         request.setHeaderField(USER_AGENT, userAgentValue);
         request.setHeaderField(ACCEPT, ACCEPT_VALUE);
         request.setHeaderField(CONTENT_TYPE, ACCEPT_VALUE + "; " + ACCEPT_CHARSET);
+        request.setHeaderField(CONTENT_LENGTH, payload != null ? String.valueOf(payload.length) : "0");
```

Comment:
Do we not have any e2e tests for this change?

Reply:
Our e2e tests already hit both the device and service side of this change. The fact that I've stopped seeing those 411 errors is about as close as I can get to verifying that this works.